java/src/jmri/jmrit/logix/NXFrame.java

```diff
@@ -294,82 +301,81 @@ private JPanel makeAutoRunPanel() {
-                _maxSpeedBox.setText(formatter.format(maxNum * _scale));
+                _maxSpeedBox.setText(formatter.format(maxNum));
                 _maxThrottleBox.setText(formatter.format(100));
                 return;
             }
-            if (_speedUnits.getText().equals(LABEL_MPH)) {
-                num = num * 0.44704f;
-            } else {
-                num = num * 0.277778f;
+            // convert num display to actual track speed in meters/sec
+            switch (_displayPref) {
+                case MPH:
+                    num = num * 0.44704f / _scale;
+                    break;
+                case KPH:
+                    num = num * 0.277778f / _scale;
+                    break;
+                case MMPS:
+                    num = num / 1000;
+                    break;
+                default:
+                    num = num / 39.37f;
+                    break;
             }
             float throttle = _speedUtil.getThrottleSettingForSpeed(num)*100;
             _maxThrottleBox.setText(formatter.format(throttle));
         });
+
         _speedUnits.addActionListener((ActionEvent evt)-> {
             NumberFormat formatter = NumberFormat.getNumberInstance(); 
             float num = 0;
             try {
                 num =  formatter.parse(_maxSpeedBox.getText()).floatValue();
             } catch (java.text.ParseException pe) {
+                _maxSpeedBox.setText(null);
                 return;
             }
-            if (_speedUnits.getText().equals(LABEL_MPH)) {
-                _speedUnits.setText(LABEL_KPH);
-                num = Math.round(num * 160.9344f);
-                _maxSpeedBox.setText(formatter.format(num / 100));
-            } else {
-                num = Math.round(num * 62.137119f);
-                _speedUnits.setText(LABEL_MPH);
-                _maxSpeedBox.setText(formatter.format(num / 100));
-            }
-        });
-        p1.add(makeTextAndButtonPanel(_maxThrottleBox, new JLabel(Bundle.getMessage("percent")), "MaxSpeed", null));
-        p1.add(makeTextAndButtonPanel(_maxSpeedBox, _speedUnits, "scaleSpeed", "ToolTipScaleSpeed"));
+            switch (_displayPref) {
+                case MPH:
+                    _displayPref = Display.KPH;
+                    _maxSpeedBox.setText(formatter.format(num * 1.60934f));
+                    break;
+                case KPH:
+                    _displayPref = Display.MMPS;
+                    _maxSpeedBox.setText(formatter.format(num * 0277.778f / _scale));
+                    _maxSpeedBoxLabel.setText(Bundle.getMessage("trackSpeed"));
+                    break;
+                case MMPS:
+                    _displayPref = Display.INPS;
+                    _maxSpeedBox.setText(formatter.format(num * 0.03937f));
+                    break;
+                default:
+                    _displayPref = Display.MPH;
+                    _maxSpeedBox.setText(formatter.format(num * 0.056818f * _scale));
```

Comment:
Please document these constants as well.

Reply:
There are more constants to be documented on lines 215, 218, 247, 249, 321, 324, 327, 453, 454 and 663.